### ISSUE TITLE
Try out Kevin's suggestion of filtering ruptures with small magnitudes

### ISSUE THREAD

**Issue by voj** [GNS-Science/nzshm-opensha#303]
We want to be able to remove the special code in our opensha branch `fix/rup-normalization`.

One of the problems is that we fill some MFD bins with 1e-20 and then have special misfit values of -1 that we ignore, using special code.

Kevin suggested we could simply filter out ruptures below 6.8 in order to reduce the size of our MFDs.

**Comment by voj** [comment]
They easiest way to filter out low magnitude ruptures was to create a new rupset and then filter out ruptures below 6.8 before the rupture set is written to file.

Running a 4 minute inversion on this filtered rupture set with *unmodified inversion code*:
![Screenshot 2024-03-22 104811](https://github.com/GNS-Science/nzshm-opensha/assets/3160916/ed9ee41c-8b94-4cd9-abb5-0144203e245c)

Comparing this to the [NZSHM22 version](http://nzshm22-static-reports.s3-website-ap-southeast-2.amazonaws.com/opensha/DATA/SW52ZXJzaW9uU29sdXRpb246NjMzMzY3Mw==/solution_report/resources/mfd_plot_Total_Target_MFDs.png) it looks very similar, but the solution is a bit lower around 6.9. 

All inversions I have run for this investigation use the same parameters as this NZSHM22 inversion. They also all use the filtered rupture set.

Next, I modified the inversion code to see if we could have MFDs start at 6.8.

Trying to set the `global minMag` to 6.8 lead to strange MFDs, for example:
![Screenshot 2024-03-22 100507](https://github.com/GNS-Science/nzshm-opensha/assets/3160916/8dac7af4-7dd8-4424-b9dc-cfa05f65484a)

Only setting the minMag for `targetOnFaultSupraSeisMFDs` to 6.8 and leave all other MFDs as before lead to a result that visually looks identical to the result with unmodified code. This is the MFD that required us to set bin values to 1e-20 and to set misfits to -1. I have confirmed that we don't set bin values to 1e-20, and the MFD misfits are no longer -1.

![Screenshot 2024-03-22 132952](https://github.com/GNS-Science/nzshm-opensha/assets/3160916/e1e0f3f4-7b24-48cf-8959-53a4953506c7)
